### PR TITLE
FIX: Pass category param on /tags/intersection

### PR DESCRIPTION
**Pull request by CvX**

### CODE REVIEW

**Review by davidtaylorhq — APPROVED** [review]

File: app/assets/javascripts/discourse/app/routes/tag-show.js [hunk 1]
```diff
@@ -143,7 +150,7 @@ export default DiscourseRoute.extend({
   setupController(controller, model) {
     const noSubcategories = this.noSubcategories;
 
-    this.controllerFor("tag.show").setProperties({
+    this.controllerFor(this.controllerName).setProperties({
```
Comment (ZogStriP) [inline app/assets/javascripts/discourse/app/routes/tag-show.js:153]: What's the difference between `this.controllerFor(this.controllerName)` and the `controller` argument?

File: app/assets/javascripts/discourse/app/routes/tags-intersection.js [hunk 2]
```diff
@@ -1,9 +1,20 @@
...
 // that confusion.
+export default TagShowRoute.extend({
+  controllerName: "tags.intersection",
+
+  init() {
+    this._super(...arguments);
+
+    // The only difference is support for `category` query param.
+    // Other routes include category in the route path.
+    this.set("queryParams", { ...queryParams });
```
Comment (ZogStriP) [inline app/assets/javascripts/discourse/app/routes/tags-intersection.js:17]: Would the following work?

```js
this.set("queryParams", queryParams);
```

I guess my question is: what does `{ ...queryParams }` do in this case?

File: app/assets/javascripts/discourse/app/routes/tags-intersection.js [hunk 3]
```diff
@@ -1,9 +1,20 @@
...
 
-export default TagShowRoute.extend({});
-
-// The tags-intersection route is exactly the same as the tags-show route, but the wildcard at the
+// The tags-intersection route is the same as the tags-show route, but the wildcard at the
 // end of the route (*additional_tags) will cause a match when query parameters are present,
 // breaking all other tags-show routes. Ember thinks the query params are addition tags and should
 // be handled by the intersection logic. Defining tags-intersection as something separate avoids
 // that confusion.
+export default TagShowRoute.extend({
```
Comment (s3lase) [inline app/assets/javascripts/discourse/app/routes/tags-intersection.js:9]: Any specific reason why you didn't  go for standard es6 class here?

File: app/assets/javascripts/discourse/app/controllers/tags-intersection.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import TagShowController from "discourse/controllers/tag-show";
+import { queryParams } from "discourse/controllers/discovery-sortable";
+
+export default class TagsIntersectionController extends TagShowController {
+  constructor() {
+    super(...arguments);
+
+    this.set("queryParams", [
+      ...Object.keys(queryParams),
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/app/controllers/tags-intersection.js:9]: So `queryParams` is already present in the scope here? Would it work with `this.queryParams` too? If yes maybe we use the `this` version to be more explicit.

File: app/assets/javascripts/discourse/app/routes/tags-intersection.js [hunk 5]
```diff
@@ -1,9 +1,20 @@
...
+  controllerName: "tags.intersection",
+
+  init() {
+    this._super(...arguments);
+
+    // The only difference is support for `category` query param.
+    // Other routes include category in the route path.
+    this.set("queryParams", { ...queryParams });
+    this.queryParams["categoryParam"] = { replace: true, refreshModel: true };
```
Comment (ZogStriP) [inline app/assets/javascripts/discourse/app/routes/tags-intersection.js:18]: Is `this.get("queryParams")` different than `this.queryParams`?

File: app/assets/javascripts/discourse/tests/acceptance/tags-intersection-test.js [hunk 6]
```diff
@@ -40,4 +39,28 @@ acceptance("Tags intersection", function (needs) {
...
+          tags: [
+            { id: 1, name: "second", topic_count: 1 },
+            { id: 2, name: "first", topic_count: 1 },
+          ],
+        },
+      });
+    });
+
+    await visit("/tags/intersection/sour/tangy?category=fruits");
+    assert.verifySteps(["request"]);
```
Comment (ZogStriP) [inline app/assets/javascripts/discourse/tests/acceptance/tags-intersection-test.js:64]: What does `verifySteps` do? 🤔

**Comment by AndrewPrigorshnev** [comment]
A general comment, having one simple system spec for such a feature would be great.

File: app/assets/javascripts/discourse/app/routes/tag-show.js [hunk 7]
```diff
@@ -150,7 +157,7 @@ export default class TagShowRoute extends DiscourseRoute {
...
```
Comment (CvX) [inline app/assets/javascripts/discourse/app/routes/tag-show.js:160]:
```suggestion
    controller.setProperties({
```

**Pull request merged**